lib/provider-registration/index.js

```diff
@@ -70,6 +71,36 @@ module.exports = class ProviderRegistration {
     })
   }
 
+  registrationMetadata () {
+    return {
+      version: this.version,
+      namespace: this.namespace,
+      providerDefinedRoutes: this.providerRoutesMetadata || this.getProviderRoutesMetadata(),
+      registeredOutputs: this.registeredOutputsMetadata || this.getRegisteredOutputsMetadata()
+    }
+  }
+
+  getProviderRoutesMetadata () {
+    this.providerRoutesMetadata = this.registeredProviderRoutes.map(route => ({
+      path: route.registeredPath,
+      methods: route.methods
+    }))
```

Comment:
This can be moved to where the `this.registeredProviderRoutes` is updated. So this function can be a simple getter.

Reply:
Updated as requested.